specification/security_model.adoc

```diff
@@ -1,9 +1,195 @@
+
+1. The TVM confidential data, which includes its confidential main memory, code
+   and state.
+2. Any assigned device I/O space that is mapped to the TVM address space,
+   also known as device MMIO.
+3. System memory.
 
 === Adversary Model
 
-=== Scope and Threat Model
+The CoVE-IO security model aims at protecting the above-described TVM assets
+from the following adversaries:
+
+* _Privileged host software adversary_: This includes host software components
+  executing in S and M mode like the host firmware, kernel, hypervisor, VMM,
+  etc. As the system resource owner but also the TVMs lifecycle manager, those
+  components can access and control all devices on the system.
+
+* _Unprivileged host software adversary_: (HS mode??)
+
+* _Privileged guest software adversary_: (Other TVM guest kernels)
+
+* _Device firmware adversary_: (Other system firmware, DMA into unassigned TVMs)
+
+* _Hardware adversary_: This includes adversaries that have unlimited physical
+  access to the system and its devices, and can use simple or advanced hardware
+  attacks such as bus interposers, voltage/clock glitchers, microscopes, power
+  usage analyzer, etc. in order to access or tamper with the protected assets.
+
+=== Threats
+
+==== CoVE-IO-T001
+
+.CoVE-IO-T001
+|===
+| Asset | Threat | Adversary | Scope | Result
+
+| TVM confidential data
+| Disclosure
+| Privileged host software
+| In scope
+| Host component reads TVM confidential data
+
+5+| **Description** - A privileged host software component programs a TDI that
+    is assigned to a TVM. By accessing the device MMIO space, the host component
+    can program direct memory access destination addresses to its own address
+    space. Device generated data intended to be copied to the TVM is accessed by
+    the host component instead.
+
+|===
+
+==== CoVE-IO-T002
+
+.CoVE-IO-T002
+|===
+| Asset | Threat | Adversary | Scope | Result
+
+| TVM confidential data
+| Tamper
+| Privileged host software
+| In scope
+| Host component writes to TVM confidential memory
+
+5+| **Description** - A privileged host software component programs a TDI that
+    is assigned to a TVM. By accessing the device MMIO space, the host component
+    can modify the programmed direct memory access destination addresses, and
+    redirect it to unintended parts of the TVM address space.
+
+|===
+
+==== CoVE-IO-T003
+
+.CoVE-IO-T003
+|===
+| Asset | Threat | Adversary | Scope | Result
+
+| TVM confidential data
+| Denial of service
+| Privileged host software
+| **Not** in scope
+| TVM can not access a TDI that is assigned to it
+
+5+| **Description** - A privileged host software component resets or powers down
+    an assigned TDI or its physical device, while the TDI is assigned to a TVM.
+    The TVM is no longer able to directly access its assigned TDI.
+
+|===
+
+==== CoVE-IO-T004
+
+.CoVE-IO-T004
+|===
+| Asset | Threat | Adversary | Scope | Result
+
+| System memory
```

Comment:
I am not sure if we want to list it here.
It seems irrelevant to confidential computing.

Reply:
It is a valid threat for any directly assigned device, not only for confidential computing, I agree. I am calling it out here as this is a threat that is not covered by the CoVE threat model.

Reply:
I agree that this is something need to be considered in a full system design.
I suggest we don't mix them with threat in confidential computing.

The reason is that: If we want to add all threat for the full system, then we need list more. for example:
hypervisor need prevent attack from OS.
OS need prevent attack from APP.
hypervisor need prevent attack from BIOS.
OS need prevent attack from BIOS.
hypervisor need prevent attack from device DMA.
OS need prevent attack from device DMA.
BIOS need prevent attack from device DMA.
......

I am not sure what is criteria to list or not list them?